From f822b3d2a0343b89c4d8673edb740e35ca8338fb Mon Sep 17 00:00:00 2001
From: mgotefox <manishx.gotefode@intel.com>
Date: Fri, 25 Mar 2016 16:13:10 -0700
Subject: [PATCH 07/10] atomisp2: css2400 and 2401
 irci_ecr-master_20160325_1500

Merge CSS2.1 for css2400 and 2401_legacy for
irci_ecr-master_20160325_1500 firmware tag

Change-Id: I7a31761b605ca43d71a676cae68b53db0baca38e
Signed-off-by: mgotefox <manishx.gotefode@intel.com>
Original-Tracked_On:
Reviewed-on: https://icggerrit.ir.intel.com/110341
Build: Dessert M BXT RVP (SC)
Build: Dessert M BTNS (SC)
Build: Dessert M CHT FFD (SC)
Approver: WeiX Meng <weix.meng@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-26488
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/487153
---
 .../drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c  | 3 ++-
 .../external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h   | 2 +-
 .../camera/drivers/media/pci/atomisp2/css/ia_css_types.h              | 2 +-
 .../camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h       | 2 +-
 .../media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h   | 2 +-
 .../camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c           | 4 ++--
 6 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c
index 5de33f8..1815180 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c
@@ -763,7 +763,8 @@ void ia_css_pipe_get_post_gdc_binarydesc(
 static unsigned int pre_de_isp27_modes[MAX_NUM_PRE_DE_STAGES] =
 {
 	IA_CSS_BINARY_MODE_PRE_DE,
-	IA_CSS_BINARY_MODE_PRE_DE_STAGE1
+	IA_CSS_BINARY_MODE_PRE_DE_STAGE1,
+	IA_CSS_BINARY_MODE_PRE_DE_STAGE2
 };
 
 void ia_css_pipe_get_pre_de_binarydesc(
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
index 078a0db..b9a3119 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20160323_0200 */
+/* Release Version: irci_ecr-master_20160325_1500 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
index 303ecd5..c44f704 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20160323_0200 */
+/* Release Version: irci_ecr-master_20160325_1500 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
index 7f935c8..e3adfc5 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
@@ -22,7 +22,7 @@ more details.
 #define __IA_CSS_VERSION_DATA_H
 
 
-#define CSS_VERSION_STRING "REL:20160323_12.3_0824; API:2.1.22.2; GIT:irci___#68ed6f17feea6cf4c24f5d25684588f180ca6bfe; SDK:; USER:viedifw; "
+#define CSS_VERSION_STRING "REL:20160325_12.5_1933; API:2.1.22.2; GIT:irci___#35298f51b665ebcc2ebd2705f9bb92d7867be639; SDK:; USER:viedifw; "
 
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h
index b4cfd6b..54ab384 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h
@@ -52,7 +52,7 @@ more details.
 #define MAX_NUM_PRIMARY_STAGES NUM_PRIMARY_HQ27_STAGES
 
 #define NUM_PRE_DE_STAGES       1 /* number of pre-de binaries can be loaded on non-ISP2.7 pipe */
-#define NUM_PRE_DE_ISP27_STAGES 2 /* number of pre-de binaries can be loaded on ISP2.7 pipe */
+#define NUM_PRE_DE_ISP27_STAGES 3 /* number of pre-de binaries can be loaded on ISP2.7 pipe */
 #define MAX_NUM_PRE_DE_STAGES   NUM_PRE_DE_ISP27_STAGES
 
 /* Indicate where binaries can read input from */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
index 44d0e80..74416e8 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
@@ -47,10 +47,10 @@ struct fw_param {
 /* Warning: same order as SH_CSS_BINARY_ID_* */
 static struct firmware_header *firmware_header;
 
-/* The string STR(irci_ecr-master_20160323_0200) is a place holder
+/* The string STR(irci_ecr-master_20160325_1500) is a place holder
  * which will be replaced with the actual RELEASE_VERSION
  * during package generation. Please do not modify  */
-static const char *release_version = STR(irci_ecr-master_20160323_0200);
+static const char *release_version = STR(irci_ecr-master_20160325_1500);
 
 #define MAX_FW_REL_VER_NAME	300
 static char FW_rel_ver_name[MAX_FW_REL_VER_NAME] = "---";
-- 
1.9.1

